### DISCUSSION TITLE
[Suggestions] Save The Game During Combat

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#7573] [Ideas]
I wish I could save the current game during a combat, and be able to reload the save file and come back to that point in the combat.
I like to do some combats in different ways and that would save me from having to start the combat all over again.
It would also be really helpful to check for bugs, etc.

The second thing I would also like to do is to be able to save and replay some combats:

"- new Script Action System (SAS) to save and replay your most epic combats."

The combat is simply saved in a text file, easily readable and small in size, showing all the actions performed during a combat from start to finish, starting armies, casting spells, moving and attacking troops, etc.

When this file is loaded in the game, there is the possibility to choose the execution mode, the first mode reproduces only the saved combat, so it's just a replay of the combat.
The second mode allows to take control at the start of the combat and to play the combat with the starting armies against the AI and also to do the combat by letting the AI play for the two heroes.
It could also be very useful for bug reporting and quick testing of AI behavior during combat.

This replay file can be used with the "Battle Only" module already present in fheroes2 (other comments on this module later).

You can read the discussion on this here:
https://heroes2.forumactif.com/t931-fheroes2-save-the-game-during-combat#8445

**Comment by ihhub** [comment]
I don't think that we should even implement this feature. The amount of work needed to do this is very high while the usefulness of this change is little small. Battles in Heroes 2 are much faster compare to older installments in the series so this possible feature would save only at max few minutes of player time.

On top of this a part of this proposal was covered by introducing instant battles. Also for such a feature a UX mockup is needed. I am requalifying this as a discussion till it reaches a maturity state.
